### DISCUSSION TITLE
How do most users add WebSocket support?

### DISCUSSION THREAD

**Discussion by bcalco** [dotnet/WatsonWebserver#79] [Q&A]
Just curious how users of WatsonWebserver add WebSocket support to their web server?

I'm interested in Watson because of its simplicity, but one thing I really need is websocket support. Also, Swagger support is a plus.

**Comment by jchristn** [comment]
Hi @bcalco this is an excellent question.  I'll let others chime in on how they do it, and I'll answer with my perspective based on multi-modal (RESTful plus websocket) systems I've had to build and my reasoning on why websocket capabilities don't exist in this library.

One of the main reasons I haven't integrated websocket capabilities into Watson is because the consumption model is much different.  With RESTful HTTP, it's generally one-and-done, i.e. a request is handled, a response is sent, and you move on to the next one.  Websockets tend to be more interactive and RPC-like.  From a consumption model perspective, i.e. the interfaces exposed by the library, they're quite different, which has been the main reason I haven't integrated Watson with [WatsonWebsocket](https://github.com/jchristn/watsonwebsocket) (another library I manage).  

So what do I do for my own multi-modal projects?  I use both libraries.  Same code base, and share logic where appropriate.  Two separate listeners/ports.  

Re: Swagger, OpenAPI, or other platforms, I'd love some ideas on how you feel support for these should/could be added.  

Cheers, Joel
